packages/material-ui-lab/src/SliderStyled/SliderStyled.js

```diff
@@ -0,0 +1,329 @@
+    left: 'calc(-50% - 4px)',
+  },
+  '& .MuiSlider-mark': {
+    position: 'absolute',
+    width: 2,
+    height: 2,
+    borderRadius: 1,
+    backgroundColor: 'currentColor',
+    '&.MuiSlider-markActive': {
+      backgroundColor: props.theme.palette.background.paper,
+      opacity: 0.8,
+    },
+  },
+  '& .MuiSlider-markLabel': {
+    ...props.theme.typography.body2,
+    color: props.theme.palette.text.secondary,
+    position: 'absolute',
+    top: 26,
+    transform: 'translateX(-50%)',
+    whiteSpace: 'nowrap',
+    ...(props.state.orientation === 'vertical' && {
+      top: 'auto',
+      left: 26,
+      transform: 'translateY(50%)',
+    }),
+    '@media (pointer: coarse)': {
+      top: 40,
+      ...(props.state.orientation === 'vertical' && {
+        left: 31,
+      }),
+    },
+    '&.MuiSlider-markLabelActive': {
+      color: props.theme.palette.text.primary,
+    },
+  },
+}));
+
+SliderRoot.propTypes = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit the d.ts file and run "yarn proptypes"     |
+  // ----------------------------------------------------------------------
+  /**
+   * @ignore
+   */
+  children: PropTypes.node,
+  /**
+   * @ignore
+   */
+  state: PropTypes.shape({
+    'aria-label': PropTypes.string,
+    'aria-labelledby': PropTypes.string,
+    'aria-valuetext': PropTypes.string,
+    classes: PropTypes.object,
+    color: PropTypes.oneOf(['primary', 'secondary']),
+    defaultValue: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    disabled: PropTypes.bool,
+    getAriaLabel: PropTypes.func,
+    getAriaValueText: PropTypes.func,
+    isRtl: PropTypes.bool,
+    marks: PropTypes.oneOfType([
+      PropTypes.arrayOf(
+        PropTypes.shape({
+          label: PropTypes.node,
+          value: PropTypes.number.isRequired,
+        }),
+      ),
+      PropTypes.bool,
+    ]),
+    max: PropTypes.number,
+    min: PropTypes.number,
+    name: PropTypes.string,
+    onChange: PropTypes.func,
+    onChangeCommitted: PropTypes.func,
+    orientation: PropTypes.oneOf(['horizontal', 'vertical']),
+    scale: PropTypes.func,
+    step: PropTypes.number,
+    track: PropTypes.oneOf(['inverted', 'normal', false]),
+    value: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    valueLabelDisplay: PropTypes.oneOf(['auto', 'off', 'on']),
+    valueLabelFormat: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+  }),
+};
+
+const getComponentProps = (components, componentsProps, name) => {
+  const slotProps = componentsProps[name] || {};
+  return {
+    as: components[name],
+    ...slotProps,
+  };
+};
+
+const Slider = React.forwardRef(function Slider(inputProps, ref) {
+  const props = useThemeProps({ props: inputProps, name: 'MuiSlider' });
+  const {
+    // TODO: these proptypes were not generated
```

Comment:
@eps1lon for some reason these were not generated (basically only children prop was added - https://github.com/mui-org/material-ui/pull/22435/files#diff-82aa664fcb0179a4414abd4ea9159578R318-R319) so I needed to add eslint-disable on the next two lines...

Reply:
That's going to be tricky because we don't know whether these props come from `inputProps` or `useThemeProps`. I'm a bit surprise that the lint rule is even flagging these. I've got some ideas it's getting into flow analysis territory which isn't all that nice in JS.